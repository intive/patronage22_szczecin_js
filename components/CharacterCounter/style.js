import styled from 'styled-components'

export const CounterContainer = styled.span`
  font-size: 0.875rem;
  color: rgb(81, 81, 81);
  margin-top: 0.5rem;
`
