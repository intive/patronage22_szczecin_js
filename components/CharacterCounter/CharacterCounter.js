import { CounterContainer } from './style'

export default function CharacterCounter ({ inputValue, maxCharacters, className }) {
  return <CounterContainer className={className} data-testid='character-counter'>{(inputValue || '').length}/{maxCharacters}</CounterContainer>
}
