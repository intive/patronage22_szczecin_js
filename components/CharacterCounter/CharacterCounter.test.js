/* globals describe, expect, it */

import { render, screen } from '@testing-library/react'
import CharacterCounter from './CharacterCounter'

describe('CharacterCounter', () => {
  it('should render CharacterCounter component', () => {
    const { container } = render(<CharacterCounter maxCharacters='30' />)

    expect(container).toMatchSnapshot()
  })

  it('should render CharacterCounter component with text 9/30', () => {
    render(<CharacterCounter inputValue='Test test' maxCharacters='30' />)

    expect(screen.getByText('9/30')).toBeTruthy()
  })

  it('should render CharacterCounter component with text 0/30 when inputValue is empty', () => {
    render(<CharacterCounter maxCharacters='30' />)

    expect(screen.getByText('0/30')).toBeTruthy()
  })
})
