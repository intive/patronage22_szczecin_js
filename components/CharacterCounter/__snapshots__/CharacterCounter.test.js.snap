// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterCounter should render CharacterCounter component 1`] = `
<div>
  <span
    class="sc-bdvvtL eePQfO"
    data-testid="character-counter"
  >
    0
    /
    30
  </span>
</div>
`;
